### PR TITLE
Pod Store

### PR DESCRIPTION
**Pull request by smallbraingames**
- sqlite pod store
- pod count in header
- future proof message types
- add schnorr sig message type & identity

note: part of this involves serializing and deserializing pods to store in db; breaking changes to pod will change ser/deserialization. I think this is ok for now we'll rug pods for a bit while this aligns. soon we can find fixes similar to the message type fix but that's stress for a biiiit later

### CODE REVIEW

File: chat/Cargo.toml [hunk 1]
```diff
@@ -15,6 +15,7 @@ gpui.workspace = true
 iroh = "0.26.0"
 pkarr = "2.2.0"
 postcard = "1.0.10"
+pod2 = { path = "../pod2" }
```
Comment (ludns) [inline chat/Cargo.toml:18]: Can you import it from the workspace? See how I did it for the "assets" crate

File: chat/Cargo.toml [hunk 2]
```diff
@@ -25,6 +26,10 @@ tracing = "0.1.40"
 tracing-appender = "0.2.3"
 tracing-subscriber = { version = "0.3.18", features = ["env-filter"] }
 unicode-segmentation = "1.12.0"
+rand = "0.8.5"
+rusqlite = { version = "0.32.1", features = ["bundled"] }
+blake3 = "1.5.4"
+plonky2 = { git = "https://github.com/mir-protocol/plonky2" }
```
Comment (ludns) [inline chat/Cargo.toml:32]: We should move this to the workspace too so we can pin it to this specific fork

File: chat/src/logic/message.rs [hunk 3]
```diff
@@ -65,7 +73,11 @@ impl Message {
     pub fn timestamp(&self) -> &DateTime<Utc> {
         match self {
             Message::AboutMe { timestamp, .. } => timestamp,
+            Message::SchnorrKey { timestamp, .. } => timestamp,
             Message::Chat { timestamp, .. } => timestamp,
+            Message::Unknown => {
+                panic!("unknown message type")
```
Comment (ludns) [inline chat/src/logic/message.rs:79]: We prob don't want to panic here? Otherwise older version of the app will panic on load and never update.

**Review by ludns — CHANGES_REQUESTED** [review]
small feedback on crate stuff and a `panic`. Great work!

**Comment by ludns** [comment]
Also should we bump version and ship an update?

File: chat/Cargo.toml (hunk 1 above)
Comment (smallbraingames) [reply chat/Cargo.toml:18]: yes

File: chat/Cargo.toml (hunk 2 above)
Comment (smallbraingames) [reply chat/Cargo.toml:32]: eh I'll just pin it to the commit hash

File: chat/src/logic/message.rs (hunk 3 above)
Comment (smallbraingames) [reply chat/src/logic/message.rs:79]: true

File: chat/Cargo.toml (hunk 2 above)
Comment (smallbraingames) [reply chat/Cargo.toml:32]: oh, I see what ur saying

like we move the dep to the workspace

Comment (smallbraingames) [reply chat/Cargo.toml:32]: yes agreed thats better

**Pull request merged**